packages/flame/lib/src/effects2/simple_effect_controller.dart

```diff
@@ -0,0 +1,54 @@
+import 'effect_controller.dart';
+import 'standard_effect_controller.dart';
+
+/// Simplest possible [EffectController], which supports an effect progressing
+/// linearly over [duration] seconds.
+///
+/// The [duration] can be 0, in which case the effect will jump from 0 to 1
+/// instantaneously.
+///
+/// The [delay] parameter allows to delay the start of the effect by the
+/// specified number of seconds.
+///
+/// See also: [StandardEffectController]
+class SimpleEffectController extends EffectController {
+  SimpleEffectController({
+    this.duration = 0.0,
+    this.delay = 0.0,
+  })  : assert(duration >= 0, 'duration cannot be negative: $duration'),
+        assert(delay >= 0, 'delay cannot be negative: $delay');
+
+  final double duration;
+  final double delay;
+  double _timer = 0.0;
+
+  @override
+  bool get started => _timer >= delay;
+
+  @override
+  bool get completed => _timer >= delay + duration;
```

Comment:
A note for later, when creating the `SequenceEffect` we need to carry over the time that overflowed in here to the next effect.

Reply:
noted